.sql/data.sql

```diff
@@ -0,0 +1,246 @@
+-- phpMyAdmin SQL Dump
+-- version 4.6.6deb1+deb.cihar.com~xenial.2
+-- https://www.phpmyadmin.net/
+--
+-- Host: localhost
+-- Czas generowania: 24 Paź 2018, 00:34
+-- Wersja serwera: 5.7.20-0ubuntu0.16.04.1
+-- Wersja PHP: 7.2.11-2+ubuntu16.04.1+deb.sury.org+1
+
+SET SQL_MODE = "NO_AUTO_VALUE_ON_ZERO";
+SET time_zone = "+00:00";
+
+--
+-- Baza danych: `adfind`
+--
+
+-- --------------------------------------------------------
+
+--
+-- Struktura tabeli dla tabeli `Category`
+--
+
+CREATE TABLE `Category` (
+  `id` int(11) NOT NULL,
+  `name` varchar(255) NOT NULL
+) ENGINE=InnoDB DEFAULT CHARSET=utf8;
+
+--
+-- Zrzut danych tabeli `Category`
+--
+
+INSERT INTO `Category` (`id`, `name`) VALUES
+(1, 'Photos'),
+(2, 'Pictograms'),
+(3, 'Illustrations'),
+(4, 'Documents');
+
+-- --------------------------------------------------------
+
+--
+-- Struktura tabeli dla tabeli `File`
+--
+
+CREATE TABLE `File` (
+  `id` int(11) NOT NULL,
+  `name_add` varchar(255) NOT NULL,
+  `name_view` varchar(255) DEFAULT NULL,
+  `name_hash` varchar(255) NOT NULL,
+  `subcategory_id` int(11) DEFAULT NULL
+) ENGINE=InnoDB DEFAULT CHARSET=utf8;
+
+
+-- --------------------------------------------------------
+
+--
+-- Struktura tabeli dla tabeli `Format`
+--
+
+CREATE TABLE `Format` (
+  `id` int(11) NOT NULL,
+  `name` varchar(255) NOT NULL
+) ENGINE=InnoDB DEFAULT CHARSET=utf8;
+
+--
+-- Zrzut danych tabeli `Format`
+--
+
+INSERT INTO `Format` (`id`, `name`) VALUES
+(1, 'jpeg'),
+(2, 'png'),
+(3, 'svg'),
+(4, 'eps'),
+(5, 'txt'),
+(6, 'doc'),
+(7, 'ppt'),
+(8, 'xls');
+
+-- --------------------------------------------------------
+
+--
+-- Struktura tabeli dla tabeli `MapFormat`
+--
+
+CREATE TABLE `MapFormat` (
```

Comment:
Nie ma potrzeby trzymać mapowania plik -> format bo php magicznie wie jakiego typu jest plik ;) Służy do tego mime i jest standardowym narzędziem linuxowym. Możesz je stosować w wielu językach. Przykładowo: http://php.net/manual/en/function.finfo-file.php . W systemie plików trzymaj jednak pliki z zahashowanymi nazwami oraz z rozszerzeniem np: abcd(...).jpg bo zauważyłem że niektóre hostingi nie ogarniają nazw bez rozszerzeń ;) U nas w Shoperze pliki graficzne zawsze mają rozszerzenie jpg mimo że nimi nie muszą być. Jak skorzystasz z narzędzi developerskich Chrome to w zakładce Network możesz podglądnąć sobie ruch. Jak klikniesz jakiś dotyczący obrazka to zauważysz w sekcji Response Headers nagłówek content-type: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Type . Zawiera on taką samą wartość jak zwraca funkcja finfo_file. Możesz więc trzymać pliki z takimi samymi rozszerzeniami mimo że są innego typu. Przeglądarka się dostosuje ;)